src/libs/actions/Report.js

```diff
@@ -492,9 +534,23 @@ function addAction(reportID, text, file) {
         maxSequenceNumber: newSequenceNumber,
     });
 
+    // Generate a client id to store with the local action. Later, we
+    // will find and remove this action by referencing the action created
+    // in the server. We do this because it's not safe to assume that this
+    // action will use the next sequenceNumber. An action created by another
+    // user can overwrite that sequenceNumber if it is created before this one.
+    const temporaryReportActionID = `${Date.now()}_${Str.guid()}`;
+
+    // Add the temporary action ID to our report map so we can remove it
+    // when refetching report actions for a given report to clear out any
+    // stuck actions.
```

Comment:
This comment was hard for me to understand so I've added a suggestion, but I'm not entirely sure that it's right.
```suggestion
    // Store the temporary action ID on the report the comment was added to.
    // It will be removed later when refetching report actions in order to clear out any
    // stuck actions (ie. actions where the client never received a Pusher event, for 
    // whatever reason, from the server with the new action data).
```

Reply:
Great suggestion and much clearer thanks!